src/components/editor/metaform-editor-right-drawer-visibility.tsx

```diff
@@ -0,0 +1,338 @@
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Updates metaform field or section visibleIf parameters
+   *
+   * @param newMetaformUpdateFieldOrSection metaform field or section
+   * @param value event condition equals value (yes / no)
+   */
+  const updateFormFieldOrSectionVisiblityValues = (newMetaformUpdateFieldOrSection: MetaformField | MetaformSection, value: string) => {
+    // TODO const amountOfRadios = pendingForm.sections!.map(x => x.fields?.filter(y => y.type === MetaformFieldType.Radio)).flat().length;
+    setRequiredConditionEqualsValue(value);
+    setRequiredConditionInfoField(false);
+    if (sectionIndex === undefined) {
+      return;
+    }
+    if (sectionIndex !== undefined && fieldIndex === undefined) {
+      const updatedForm = produce(pendingForm, draftForm => {
+        draftForm.sections?.splice(sectionIndex, 1, newMetaformUpdateFieldOrSection as MetaformSection);
+      });
+      setPendingForm(updatedForm);
+    }
+    if (sectionIndex !== undefined && fieldIndex !== undefined) {
+      const updatedForm = produce(pendingForm, draftForm => {
+        draftForm.sections?.[sectionIndex]?.fields?.splice(fieldIndex, 1, newMetaformUpdateFieldOrSection as MetaformField);
+      });
+      setPendingForm(updatedForm);
+    }
+  };
+
+  /**
+   * Updates metaform section
+   *
+   * @param newMetaformSection new metaform section
+   */
+  const emptyFormSectionVisibleIf = (newMetaformSection: MetaformSection) => {
+    if (sectionIndex === undefined) {
+      return;
+    }
+
+    const section = pendingForm.sections?.[sectionIndex];
+    if (!section) {
+      return;
+    }
+
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.splice(sectionIndex, 1, newMetaformSection);
+    });
+
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Render visiblity condition selection menu
+   *
+   */
+  const fieldConditionComponent = () => {
+    if (sectionIndex !== undefined) {
+      return (
+        <>
+          <Typography variant="subtitle1" style={{ width: "100%" }}>
+            { strings.draftEditorScreen.editor.features.visiblityCondition }
+          </Typography>
+          <FormControl fullWidth disabled={ !disabledValue }>
+            <InputLabel id="fieldCategoryVisiblityConditionLabel">
+              { requiredConditionField ?
+                strings.draftEditorScreen.editor.visibility.conditionLabelTitle :
+                strings.draftEditorScreen.editor.visibility.fieldDefiningCondition
+              }
+            </InputLabel>
+            <Select
+              fullWidth
+              labelId="fieldCategoryVisiblityConditionLabel"
+              label={ requiredConditionField ?
+                strings.draftEditorScreen.editor.visibility.conditionLabelTitle :
+                strings.draftEditorScreen.editor.visibility.fieldDefiningCondition
+              }
```

Comment:
Since the same string is being used in two places, this ternary check should propably be moved to its own variable and use the value of that variable in these two places to clean and simplify this.
Something like:
`const labelText = requiredConditionField ? strings.draftEditorSCreen.editor.visiblity.conditionLabelTitle : strings.draftEditorScreen.editor.visibility.fieldDefiningCondition`